presentation/src/main/java/com/exallium/mvvmcleanexample/presentation/utils/delegates/EcoDisposableProperty.kt

```diff
@@ -0,0 +1,15 @@
+package com.exallium.mvvmcleanexample.presentation.utils.delegates
+
+import io.reactivex.disposables.Disposable
+import kotlin.properties.ReadWriteProperty
+import kotlin.reflect.KProperty
+
+class EcoDisposableProperty(private var disposable: Disposable?) : ReadWriteProperty<Any?, Disposable?> {
```

Comment:
I'm even thinking of like a CompositeDisposableDelegate that you just add things to, and when you set it to null it clears it.  Not really sure what that'd look like.  in fact, it'd probably be easier to just use a CompositeDisposable.